src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -757,16 +754,13 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
 
 #endif
             }
-            else if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-            {
-                ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));
-            }
 
             GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_ZEROING_OPTIONAL;
             if (pinned)
                 flags |= GC_ALLOC_FLAGS.GC_ALLOC_PINNED_OBJECT_HEAP;
 
-            return Unsafe.As<T[]>(AllocateNewArray(RuntimeTypeHandle.ToIntPtr(typeof(T[]).TypeHandle), length, flags));
+            // Runtime overrides GC_ALLOC_ZEROING_OPTIONAL if the type contains references, so we don't need to worry about that.
+            return Unsafe.As<T[]>(AllocateNewArray(typeof(T[]).TypeHandle.Value, length, flags));
```

Comment:
Why did you change `RuntimeTypeHandle.ToIntPtr` to `RuntimeTypeHandle.Value`?

`RuntimeTypeHandle.ToIntPtr` is JIT intrinsic so it will be optimized to just loading a constant in most cases. `RuntimeTypeHandle.Value` does not have that optimization. This change may cause a perf regression.

Reply:
> Oh! I had no idea it was an intrinsic, how is that discoverable?

JIT intrinsics are marked with `[Intrinsic]` attributes.

> Perhaps the should instead revert this line and change the other to use the intrinsic?

Agree.

Reply:
Ok, I'll do the change once I get closure on the other discussion